pkg/arvo/neo/cod/std/src/fil/tasks.md

```diff
@@ -0,0 +1,613 @@
+    %-  sort  :_  lth
+    %+  turn  ~(tap by ~(tar of:neo kids.bowl))
+      |=  [=pith =idea:neo]
+      +:(,[%ud @ud] (rear pith))
+  =/  last-name=@ud  (rear sorted-names)
+  =/  name-missing=(list @ud)
+    %+  skim  (gulf 1 last-name)
+    |=  n=@ud
+    =(~ (find ~[n] sorted-names))
+  ?~  name-missing  +(last-name)
+  (rear name-missing)
+--
+::
+::  outer core
+^-  kook:neo
+|%
+::
+::  state is a %task
+++  state  pro/%task
+::
+::  we accept %task-diff and %gift pokes
+++  poke   (sy %task-diff %gift ~)
+::
+::  we define one generation of
+::  kids at /path/to/this/task/<@ud>
+++  kids
+  :+  ~  %y
+  %-  ~(gas by *lads:neo)
+  :~  :-  [|/%ud |]
+    [pro/%task (sy %task-diff %gift ~)]
+  ==
+++  deps  *deps:neo
+++  form
+  ::
+  ::  inner core
+  ^-  form:neo
+  ::  XX refactor sample to same as other tutorials
+  |_  [=bowl:neo =aeon:neo stud:neo state-vase=vase]
+  ::  XX refactor this to default +init from counter
+  ++  init
+    |=  pal=(unit pail:neo)
+    ^-  (quip card:neo pail:neo)
+    `(need pal)
+  ++  poke
+    |=  [=stud:neo vax=vase]
+    ^-  (quip card:neo pail:neo)
+    =/  this  !<(task state-vase)
+    ?+    stud  !!
+        %gift
+      ?:  (check-kids bowl)
+        [~ task/!>(this(done %.y))]
+      [~ task/!>(this(done %.n))]
+    ::
+        %task-diff
+      =/  diff  !<(task-diff vax)
+      ?-    -.diff
+          %new
+        =/  name=@ud  (assign-name bowl)
+        =.  order.this
+          ?:  prepend.diff
+            [~[ud/name] order.this]
+          (snoc order.this `pith`[ud/name ~])
+        =.  done.this  |
+        :_  task/!>(this)
+        :~  :-  (welp here.bowl ~[ud/name])
+            [%make %task `task/!>(task.diff) ~]
+        ==
+      ::
+          %edit
+        :-  ~
+        :-  %task
+        !>
+        %=  this
+          text  text.diff
+          done  ?:  (check-kids bowl)
+                  done.diff
+                %.n
+        ==
+      ::
+          %oust
+        =/  i  (find [pith.diff ~] order.this)
+        ?~  i  `task/!>(this)
+        :_  task/!>(this(order (oust [(need i) 1] order.this)))
+        :~  [(welp here.bowl pith.diff) [%tomb ~]]
+        ==
+      ::
+          %reorder
+        `task/!>(this(order order.diff))
+      ==
+    ==
+  --
+--
+```
+
+## The %gift poke
+In the `+poke` arm we declare this shrub takes a `%gift` as well as a `%task-diff`, but we don’t have to import `%gift`. This is a special poke like `%rely` that `/app/neo` gives us when another shrub’s state changes.
+
+Our shrub receives a `%gift` poke every time the state of one of its descendants changes: only kids in the `%y` case, all descendants in the `%z` case.
+
+```hoon
+++  state  pro/%task
+++  poke   (sy %task-diff %gift ~)
+```
+
+Then we can handle the poke like any other. In this case, when `/imp/task` receives word that one of its kids’ state has changed, it checks to see if all of its kids are completed, then marks its own state as completed or uncompleted accordingly.
```

Comment:
Note that we've changed this code. The result of `check-kids` now also gets stored as a variable in state. You'll need to update both the code snippets here and add a little addendum to this explanation.